addons/arsenal/missions/Arsenal.VR/initPlayerLocal.sqf

```diff
@@ -118,33 +122,45 @@ private _coverObjects = [
     "Land_VR_CoverObject_01_standHigh_F"
 ];
 
+private _coverObjectsCount = count _coverObjects;
+
+// Set up cover objects
 for "_i" from 5 to 11 do {
     private _direction = 180 + _i * 45;
     private _position = _unit getRelPos [(abs sin _direction + abs cos _direction) * 3, _direction];
-    private _block = createVehicle [_coverObjects select (_i % count _coverObjects), _position, [], 0, "NONE"];
+    private _block = createVehicle [_coverObjects select (_i % _coverObjectsCount), _position, [], 0, "NONE"];
     _block setPos _position;
 };
 
-//--- Starting point
+// Starting point
 private _square = createVehicle ["VR_Area_01_square_1x1_grey_F", position _unit, [], 0, "NONE"];
 _square setPosASL getPosASL _unit;
 
 private _marker = createMarker [QGVAR(start), getPosWorld _unit];
 _marker setMarkerType "mil_start";
 
-//--- Open Arsenal
+// Init Arsenal
 [_unit, true, false] call FUNC(initBox);
 
-[{!isNull findDisplay 46}, {
+/*
+[{!isNull findDisplay IDD_MISSION}, {
     [_this, _this, true] call FUNC(openBox);
 }, _unit] call CBA_fnc_waitUntilAndExecute;
+*/
+// Open arsenal when loading screen has finished
+["CBA_loadingScreenDone", {
```

Comment:
From my tests this works just as well, but should it be changed?

Reply:
Seems safer not to.
Or even, the old one fires sooner.
Which means arsenal is opened behind loading screen and player won't notice the short lag when it opens.